adapters/oidc/spring-security/src/test/java/org/keycloak/adapters/springsecurity/filter/KeycloakPreAuthActionsFilterTest.java

```diff
@@ -0,0 +1,90 @@
+package org.keycloak.adapters.springsecurity.filter;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.keycloak.adapters.AdapterDeploymentContext;
+import org.keycloak.adapters.KeycloakDeployment;
+import org.keycloak.adapters.NodesRegistrationManagement;
+import org.keycloak.adapters.PreAuthActionsHandler;
+import org.keycloak.adapters.spi.HttpFacade;
+import org.keycloak.adapters.spi.UserSessionManagement;
+import org.keycloak.adapters.springsecurity.filter.KeycloakPreAuthActionsFilter.PreAuthActionsHandlerFactory;
+import org.mockito.Mock;
+import org.springframework.context.ApplicationContext;
+
+public class KeycloakPreAuthActionsFilterTest {
+
+    private KeycloakPreAuthActionsFilter filter;
```

Comment:
I would have loved to not mock all those things, but `KeycloakPreAuthActionsFilter` is a very social creature, talking to lots of folks. Too much responsibility, anyone? 

To me, a less-than-optimal unit test is a sign of a class-under-test in want of refactoring. Don't blame the test, blame the class. Not religiously defending my test. Blow it away if you want. I just felt it should be covered, otherwise how would I know my change doesn't break anything.

Reply:
@fschmager, your changes looks fine and work for me. 

Regarding tests, I agree with @slaskawi. In fact, the test also fails to interact with a Keycloak server to actually make sure the node was successfully registered during deploy and unregistered during undeploy.

However, I spring security adapter was a contribution from community IIRC. We accepted the contribution without any testsuite for spring security adapter and that is something we are lacking since then. 

I'm not sure if we want to work on a new testsuite now, maybe @stianst can give us his view about this situation. So considering all that and at least the unit test you provided, I'm OK about merging this set of changes.